### PR TITLE
Update the documentation about the maybe feature

### LINKED ISSUE #6950: documentation: `maybe` missing from list of Reserved Words
**Describe the bug**
`maybe` is not considered a Reserved Word in https://www.erlang.org/doc/reference_manual/introduction.html#reserved-words.

**Expected behavior**
`maybe` should make that list.

**Affected versions**
26 (and maybe 25?).

**Additional context**
I didn't check for other words that might be missing. Is there an easy reference we can check?

### PR DESCRIPTION
**Pull request by bjorng**
Closes #6950

### CODE REVIEW

File: system/doc/reference_manual/introduction.xml [hunk 1]
```diff
@@ -95,10 +95,15 @@
...
     <p><c>after and andalso band begin bnot bor bsl bsr bxor case catch
-      cond div end fun if let not of or orelse receive rem try
+      cond div end fun if let maybe not of or orelse receive rem try
       when xor</c></p>
     <p><strong>Note</strong>: <c>cond</c> and <c>let</c>, while reserved,
-      are currently not used by the language.</p>
+    are currently not used by the language.</p>
+    <change><p><c>maybe</c> is a reserved word only if feature
+    <c>maybe_expr</c> is enabled. In Erlang/OTP 25 and 26,
+    <c>maybe_expr</c> is disabled by default.  Starting from
```
Comment (paulo-ferraz-oliveira) [inline system/doc/reference_manual/introduction.xml:104]:
```suggestion
    <c>maybe_expr</c> is disabled by default. Starting from
```

File: system/doc/general_info/upcoming_incompatibilities.xml [hunk 2]
```diff
@@ -108,6 +108,46 @@
...
+  <section>
+    <title>OTP 27</title>
+
+    <section>
+      <title>Feature maybe_epxr will be enabled by default</title>
+      <p>
+	As of OTP 27, the <c>maybe_expr</c> feature will be approved
+	and enabled by default. That means that code that uses the
+	unquoted atom <c>maybe</c> will fail to compile. All uses of
+	<c>mabye</c> as an atom will need to be quoted. Alternatively, as a
```
Comment (paulo-ferraz-oliveira) [inline system/doc/general_info/upcoming_incompatibilities.xml:122]:
```suggestion
	<c>maybe</c> as an atom will need to be quoted. Alternatively, as a
```

File: system/doc/general_info/upcoming_incompatibilities.xml [hunk 3]
```diff
@@ -108,6 +108,46 @@
...
+
+  <section>
+    <title>OTP 29</title>
 
+    <section>
+      <title>It will no longer be possible to disable feature maybe_epxr</title>
+      <p>
+	As of OTP 29, the <c>maybe_expr</c> feature will become
+	permanent and no longer possible to disable. All uses of
+	<c>mabye</c> as an atom will need to be
```
Comment (paulo-ferraz-oliveira) [inline system/doc/general_info/upcoming_incompatibilities.xml:142]:
```suggestion
	<c>maybe</c> as an atom will need to be
```

**Comment by bjorng** [comment]
Thanks for noticing the typos. Fixed.

**Pull request merged**